src/components/Nav/Nav.js

```diff
@@ -1,10 +1,114 @@
 import React from 'react';
+import { Link, withRouter } from 'react-router-dom';
+import Button from '../Button/Button';
+import Header from '../Header/Header';
 import './Nav.scss';
 
 class Nav extends React.Component {
+  constructor() {
+    super();
+    this.state = { isHamburgerClicked: false, userLists: {} };
+    this.hamburgerRef = React.createRef();
+  }
+
+  hadleUserData() {
+    fetch('/data/userData.json', {
+      method: 'GET',
+    })
+      .then(res => res.json())
+      .then(data =>
+        this.setState({
+          userLists: data,
+        })
+      );
+  }
+
+  goToPlace = link => {
+    const { history } = this.props;
+    history.push(`${link}`);
+  };
+
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleClickOutside);
+    this.hadleUserData();
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+  }
+
+  handleClickOutside = event => {
+    if (this.hamburgerRef.current.contains(event.target)) {
+      this.setState({
+        isHamburgerClicked: true,
+      });
+    } else {
+      this.setState({
+        isHamburgerClicked: false,
+      });
+    }
+  };
```

Comment:
```suggestion
  handleClickOutside = event => {
    const condition = this.hamburgerRef.current.contains(event.target);
    this.setState({
      isHamburgerClicked: condition ? true : false,
    });
  };
```
이런식으로 축약할 수 있습니다!
setState 시 조건도 같이 넣어줘도 무방합니다만 가독성을 위해 변수로 분리 후 사용하시는 편이 가장 좋습니다!

Reply:
도현님.... 항상 좋은방법 알려주셔서 감사합니다 잘 인지하도록 계속해서 노력하겠습니다 !